### DISCUSSION TITLE
How to add display name to node id?

### DISCUSSION THREAD

**Discussion by TheFern2** [FreeOpcUa/opcua-asyncio#992] [Q&A] [answered]
Just like the title says, if you see on the pic below Rockwell FT, Ignition, and Kepware all add the display name with the initial string namespace identification, is this possible with the Server class?

![image](https://user-images.githubusercontent.com/10265682/183111116-6dda1721-1782-4b8e-8c37-81ccadc1462d.png)

**Comment by oroulet** [comment]
Asyncua is a programming library. Call the functions you want and print result as you want.
And if you have an issue show the code not behaving as you expect

**Reply by TheFern2** [reply]
Thanks, I've already found the answer, is this documented anywhere? I'll be happy to submit a PR.

**Comment by TheFern2** [comment, marked as answer]
For nodes to include display name you can do it with an f string as follows.

```
n_str=f"ns={idx};s=MyVariable"

myvar = await myobj.add_variable(n_str, "MyVariable" , 0.0)
```
